### ISSUE TITLE
youtube/VideosInsertCall.Do - EOF error

### ISSUE THREAD

**Issue by porjo** [google/google-api-go-client#147]
When uploading a large video file (>500MB) to Youtube I'm getting err==EOF from [VideosInsertCall.Do](https://godoc.org/google.golang.org/api/youtube/v3#VideosInsertCall.Do) some way into the upload.

I wonder if http.Request.Close should be set as per http://stackoverflow.com/a/34474535/202311
I'm seeing reports of a bug being fixed in go 1.6 which may be related to this - http://stackoverflow.com/a/19006050/202311 (I haven't test with 1.6)

My code is here https://github.com/porjo/youtubeuploader/blob/master/youtubeuploader.go

**Comment by mcgreevy** [comment]
Does this happen reliably (every time you try to upload?  at the same point in each upload?)

Some thoughts:

Assuming that the cause is the same as that in the linked stackoverflow reports:

* the fix in 1.6 (https://go-review.googlesource.com/#/c/3210) will not help here, as that only retries known-idempotent requests (determined by whether the http method is of an appropriate type, such as GET but not POST)

* if this is happening some way into the upload, the only requests being made under the hood at that point are POSTs.

@okdave  Assuming that the cause has been correctly diagnosed here, it looks like we could fix this by just retrying here: https://github.com/google/google-api-go-client/blob/master/gensupport/resumable.go#L166
Even though these are POST requests, it's safe to retry them, since they are made in the context of a resumable upload. Would you have a problem extending https://github.com/google/google-api-go-client/blob/master/gensupport/retry.go#L69 to also return true for err == io.EOF?

**Comment by porjo** [comment]
Closing this off. I still get these errors from time to time, but it is very rare now - typically something like this:
```
POST https://www.googleapis.com/upload/youtube/v3/videos?alt=json&part=snippet%2Cstatus&uploadType=resumable&upload_id=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx: unexpected EOF
```
